packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts

```diff
@@ -0,0 +1,108 @@
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { findPkgPath, updatePkg } from './util';
+
+/**
+ * Builder options
+ */
+export interface BuilderOptions {
+  /**
+   * Entry file
+   */
+  readonly entry: string;
+
+  /**
+   * The output directory
+   */
+  readonly outDir: string;
+
+  /**
+   * Expose modules as UMD under this name
+   */
+  readonly global: string;
+
+  /**
+   * Minify
+   */
+  readonly minify?: boolean;
+
+  /**
+   * Include source maps
+   */
+  readonly sourceMaps?: boolean;
+
+  /**
+   * The cache directory
+   */
+  readonly cacheDir?: string;
+
+  /**
+   * The node version to use as target for Babel
+   */
+  readonly nodeVersion?: string;
+}
+
+/**
+ * Builder
+ */
+export class Builder {
+  private readonly parcelBinPath: string;
+
+  constructor(private readonly options: BuilderOptions) {
+    const parcelPkgPath = require.resolve('parcel-bundler/package.json'); // This will throw if `parcel-bundler` cannot be found
```

Comment:
Probably want to `try-catch` and throw a nice error explaining that parcel needs to be installed